layout/AddressesUsingIdsQuery.js

```diff
@@ -104,14 +163,34 @@ function createStreetShould(vs) {
     should.bool.boost = vs.var('boost:street');
   }
 
-  return should;
+  // Give a little should boost to features that have a parent id that
+  // perfectly matches part of the searched for admin hierarchy,
+  // so they beat out hits that were retrieved based on bounding box containment
+  const layer_filters = createLayerIdsShould(vs);
+  if (!_.isEmpty(layer_filters)) {
+    should.bool.should = [...should.bool.should || [], layer_filters];
+  }
 
+  return should;
 }
 
-function createLayerIdsShould(layer, ids) {
-  // create an object initialize with terms.'parent.locality_id' (or whatever)
-  // must use array syntax for 2nd parameter as _.set interprets '.' as new object
-  return _.set({}, ['terms', `parent.${layer}_id`], ids);
+function createLayerBoundingBoxesShould(vs, bboxes, scale = 1.0) {
```

Comment:
In the case of a country which spans the antimeridian (such as [New Zealand](https://spelunker.whosonfirst.org/id/85633345/)) or with distant foreign territories (such as [France](https://spelunker.whosonfirst.org/id/85633147/)), what is the expected behaviour?

It seems like in both of those cases the bounding box will encompass a huge area of the world, which could have a negative effect on both perf and accuracy?

One solution might be to calculate multiple bounding boxes based on the exterior polygon rings and then use those instead so that the undesirable sections are not included in the query.

Reply:
On reflection, this is fairly easily solved by only targeting some less granular placetypes, it doesn't really make sense for countries anyway.

Reply:
We could even get real fancy and define a maximum `bbox` area, right? That way all our queries for stuff on the border of Rome and the Vatican can benefit from this logic :)

In all seriousness I'm not sure if that's worth the effort but it's certainly possible.

Reply:
filtered down to 
      const should_use_admin_bounding_box = ['neighbourhood', 'borough', 'locality', 'county', 'macrocounty', 'region'].includes(layer);